### ISSUE TITLE
Assertion failed in ConnectByIPAddress

### ISSUE THREAD

**Issue by Ravbug** [ValveSoftware/GameNetworkingSockets#167]
When trying to invoke `ConnectByIPAddress`, an assertion fails: `csteamnetworkingsockets.cpp(282): Assertion Failed: false`

The code I am using: 
```cpp
SteamNetworkingIPAddr ip;
ip.Clear();
if(!ip.ParseString(address.c_str())){
	Debug::Fatal("Invalid IP: {}",address);
}
ip.m_port = port;

SteamNetworkingConfigValue_t options;
options.SetPtr(k_ESteamNetworkingConfig_Callback_ConnectionStatusChanged, (void*)callback);
connection = interface->ConnectByIPAddress(ip, 1, &options);
if (connection == k_HSteamNetConnection_Invalid){
	Debug::Fatal("Cannot connect to {}:{}",address,port);
}
```
`address` is equal to `127.0.0.1` and port is `6000`
`interface` is initialized and used earlier to create a listen socket successfully. 

Context for the assertion that failed:
```cpp
//csteamnetworkingsockets.cpp

// Make sure connection is not in the process of being self-destructed
bool bLocked = false;
for (;;)
{

	// Fetch the state of the connection.  This is OK to do
	// even if we don't have the lock.
	ESteamNetworkingConnectionState s = pResult->GetState();
	if ( s == k_ESteamNetworkingConnectionState_Dead )
		break;
	if ( bForAPI )
	{
		if ( !BConnectionStateExistsToAPI( s ) )
			break;
	}
	else
	{
		if ( s == k_ESteamNetworkingConnectionState_None )
		{
			Assert( false );    //<------- asserts here
			break;
		}
	}
```
Is some initialization missing?

**Comment by ltndkl** [comment]
Same issue here on Windows 10. 
I built the project completely following the guide, tested _example_chat_ , and ended up with this assertion failure on the client and no exception on the server.

Callstack:
`SteamNetworkingSocketsLib::InternalGetConnectionByHandle()` line 283
`SteamNetworkingSocketsLib::GetConnectionByHandle()` line 314
`SteamNetworkingSocketsLib::EvaluateScopeConfigValue<void *>()` line 1618
`SteamNetworkingSocketsLib::SetConfigValueTyped<void *>()` line 1796
`SteamNetworkingSocketsLib::CSteamNetworkingUtils::SetConfigValue()` line 1958
`ISteamNetworkingUtils::SetConfigValueStruct()` line 432
`SteamNetworkingSocketsLib::CSteamNetworkConnectionBase::BInitConnection()` line 1030
`SteamNetworkingSocketsLib::CSteamNetworkConnectionUDP::BInitConnect()` line 1150
`SteamNetworkingSocketsLib::CSteamNetworkingSockets::ConnectByIPAddress()` line 971
`ChatClient::Run()` line 568

Removing the options passed in _ConnectByIPAddress_ made the example work just fine, however this can't be a solution.

**Comment by Ravbug** [comment]
This issue affects both [Release 1.2.0](https://github.com/ValveSoftware/GameNetworkingSockets/releases/tag/v1.2.0) and master branch as of [3b4787aff7de2aa0ad7dd4ce26d334df0370e2b2](https://github.com/ValveSoftware/GameNetworkingSockets/commit/3b4787aff7de2aa0ad7dd4ce26d334df0370e2b2).

**Comment by fletcherdvalve** [comment]
Seems pretty straightforward.  Will fix this next week.

**Comment by fletcherdvalve** [comment]
Fixed by 332da087582dacb6e474ebaab560b688ad3276a2
